undefined: libvirt.Connect libvirt.StorageVol

#  Version Reports:

### Distro  version of host:

Gentoo profile 17.0

### Terraform Version Report

Terraform v0.11.3

### Libvirt version

4.0.0

### terraform-provider-libvirt plugin version (git-hash)

tag v0.3
___
# Description of Issue/Question
When compiling from source the following error happens:

```
(...)
mkdir -p $WORK/github.com/dmacvicar/terraform-provider-libvirt/
cd /var/tmp/portage/app-admin/terraform-provider-libvirt-0.3/work/gopath/src/github.com/dmacvicar/terraform-provider-libvirt/libvirt
/usr/lib64/go/pkg/tool/linux_amd64/compile -o $WORK/github.com/dmacvicar/terraform-provider-libvirt/libvirt.a -trimpath $WORK -goversion go1.9.2 -p github.com/dmacvicar/terraform-provider-libvirt/libvirt -complete -buildid 3b44f425d6fc9792491ad186c1af6560cd9dcac4 -importmap github.com/davecgh/go-spew/spew=github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/davecgh/go-spew/spew -importmap github.com/hashicorp/terraform/helper/mutexkv=github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/hashicorp/terraform/helper/mutexkv -importmap github.com/hashicorp/terraform/helper/resource=github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/hashicorp/terraform/helper/resource -importmap github.com/hashicorp/terraform/helper/schema=github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/hashicorp/terraform/helper/schema -importmap github.com/hashicorp/terraform/terraform=github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/hashicorp/terraform/terraform -importmap github.com/hooklift/iso9660=github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/hooklift/iso9660 -importmap github.com/imdario/mergo=github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/imdario/mergo -importmap github.com/libvirt/libvirt-go=github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/libvirt/libvirt-go -importmap github.com/libvirt/libvirt-go-xml=github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/libvirt/libvirt-go-xml -importmap github.com/mitchellh/packer/common/uuid=github.com/dmacvicar/terraform-provider-libvirt/vendor/github.com/mitchellh/packer/common/uuid -importmap gopkg.in/yaml.v2=github.com/dmacvicar/terraform-provider-libvirt/vendor/gopkg.in/yaml.v2 -D _/var/tmp/portage/app-admin/terraform-provider-libvirt-0.3/work/gopath/src/github.com/dmacvicar/terraform-provider-libvirt/libvirt -I $WORK -I /var/tmp/portage/app-admin/terraform-provider-libvirt-0.3/work/gopath/pkg/linux_amd64 -pack ./cloudinit_def.go ./config.go ./coreos_ignition_def.go ./disk_def.go ./domain.go ./domain_def.go ./libvirt_interfaces.go ./network_def.go ./provider.go ./qemu_agent.go ./resource_cloud_init.go ./resource_libvirt_coreos_ignition.go ./resource_libvirt_domain.go ./resource_libvirt_domain_console.go ./resource_libvirt_domain_netiface.go ./resource_libvirt_network.go ./resource_libvirt_volume.go ./stream.go ./utils.go ./utils_domain_def.go ./utils_libvirt.go ./utils_net.go ./utils_volume.go ./volume_def.go
# github.com/dmacvicar/terraform-provider-libvirt/libvirt
libvirt/cloudinit_def.go:56:50: undefined: libvirt.Connect
libvirt/cloudinit_def.go:213:44: undefined: libvirt.Connect
libvirt/cloudinit_def.go:307:27: undefined: libvirt.Connect
libvirt/cloudinit_def.go:307:51: undefined: libvirt.StorageVol
libvirt/config.go:16:11: undefined: libvirt.Connect
```

I encountered the same error except that I was trying to cross-compile for macOS. Compiling for Linux, using just `go build -v github.com/dmacvicar/terraform-provider-libvirt`, churned out the terraform plugin just fine.

```
env GOOS=darwin GOARCH=amd64 go build -v github.com/dmacvicar/terraform-provider-libvirt
github.com/dmacvicar/terraform-provider-libvirt/libvirt

# github.com/dmacvicar/terraform-provider-libvirt/libvirt
github.com/dmacvicar/terraform-provider-libvirt/libvirt/cloudinit_def.go:56:50: undefined: libvirt.Connect
github.com/dmacvicar/terraform-provider-libvirt/libvirt/cloudinit_def.go:213:44: undefined: libvirt.Connect
github.com/dmacvicar/terraform-provider-libvirt/libvirt/cloudinit_def.go:307:27: undefined: libvirt.Connect
github.com/dmacvicar/terraform-provider-libvirt/libvirt/cloudinit_def.go:307:51: undefined: libvirt.StorageVol
github.com/dmacvicar/terraform-provider-libvirt/libvirt/config.go:16:11: undefined: libvirt.Connect
```

@dmacvicar Can you advise if cross-compiling for macOS is possible at this point in time? Thank you!

--- a/README.md
+++ b/README.md
@@ -29,6 +29,7 @@ servers on a [libvirt](https://libvirt.org/) host via [Terraform](https://terraf
 * latest [golang](https://golang.org/dl/) version
 * `mkisofs` is required to use the [CloudInit](website/docs/r/cloudinit.html.markdown)
   feature.
+* `cgo` is required by the [libvirt-go](https://github.com/libvirt/libvirt-go) package. `export CGO_ENABLED="1"`
 
 [Copied from the Terraform documentation](https://www.terraform.io/docs/plugins/basics.html):
 > To install a plugin, put the binary somewhere on your filesystem, then configure Terraform to be able to find it. The configuration where plugins are defined is ~/.terraformrc for Unix-like systems and %APPDATA%/terraform.rc for Windows.

